Makefile

```diff
@@ -249,6 +249,8 @@ test-e2e-local:
 #Test TestCmd Policy
 test-cmd: cli
 	$(PWD)/$(CLI_PATH)/kyverno test https://github.com/kyverno/policies/main
+	$(PWD)/$(CLI_PATH)/kyverno test https://github.com/kyverno/policies --git-branch main
```

Comment:
Can we get the current branch using the git command to test against the main and release branches? We also maintain release branches in `kyverno/policies` so let's cover tests in these branches as well.

Reply:
Sure, you mean like adding another test for the current release branch `release-1.5` or dynamically fetching the latest release branch from `kyverno/policies`? If it's the latter, then we might need to use the GitHub API to find the latest release branch from https://api.github.com/repos/kyverno/policies/branches

Reply:
> Sure, you mean like adding another test for the current release branch `release-1.5` or dynamically fetching the latest release branch from `kyverno/policies`? If it's the latter, then we might need to use the GitHub API to find the latest release branch from https://api.github.com/repos/kyverno/policies/branches

Another workaround for having a static release branch could be that we name the latest release branch as `release` and all the previous release branches by their versions, similar to `release-x.y` _(if that's what we're trying to do here)_

Reply:
The branch name is available in the Makefile: https://github.com/kyverno/kyverno/blob/b17e76493e00a729cf069a528575dabbff92fc7d/Makefile#L7.

We just need to update this command to test against the current branch instead of hard-coding the branch name. This change also needs to be backported to `release-1.5` branch in order to test 1.5.x policies.

Reply:
Sorry @realshuting, but I'm a bit puzzled here. Please correct me if I'm wrong.

The `GIT_BRANCH` declaration in Makefile refers to the branch of repo `kyverno/kyverno` since the Makefile itself exists in the folder where `kyverno/kyverno` repo is initialized. Due to this, whichever git command we execute in the Makefile would refer to `kyverno/kyverno`, but the repo on which we're trying to perform our tests here is `kyverno/policy`.
In a scenario where the user tries to checkout to a temporary branch and runs this test while using `GIT_BRANCH` or any other git command implementation as a value for `--git-branch`, it would immediately fail because 1. that branch would refer to `kyverno/kyverno`, not the git repo provided in the CLI, and 2. being a local temporary branch it does not really exist in the remote repository.
So in order to make it work with the remote without hard-coding the branches we'd need to fetch and use the latest branch names from remote for the provided git repository _(`kyverno/policies` in this case)_ during the test execution.

Reply:
My question is: with CLI tests automated in kyverno/kyverno `release-1.5` branch, how can we test sample policies from `release-1.5` kyverno/policies?

Reply:
Hey @realshuting, If we're going to substitute `GIT_BRANCH` for testing policies on the current branch, I propose adding a conditional in `Makefile` to test `kyverno/policies` on the current release branch in case the current branch has `release-*` pattern in order to avoid failures due to possible differences in policy structures between release and main branches, and on the `main` branch in all other cases.
I'd like to know if this is acceptable, and if not, then what alternatives could be approached to prevent similar testing failures.